src/test/lib/logical_query_plan/union_node_test.cpp

```diff
@@ -259,4 +256,79 @@ TEST_F(UnionNodeTest, UniqueColumnCombinationsUnionAll) {
   EXPECT_TRUE(union_node->unique_column_combinations().empty());
 }
 
+TEST_F(UnionNodeTest, OrderDependenciesUnionPositions) {
+  const auto od_a_to_b = OrderDependency{{_a}, {_b}};
+  const auto order_constraint = TableOrderConstraint{{ColumnID{0}}, {ColumnID{1}}};
+  _mock_node1->set_order_constraints({order_constraint});
+  EXPECT_EQ(_mock_node1->order_dependencies().size(), 1);
+
+  // Forward OD.
+  const auto& order_dependencies = _union_node->order_dependencies();
+  EXPECT_EQ(order_dependencies.size(), 1);
+  EXPECT_TRUE(order_dependencies.contains(od_a_to_b));
+}
+
+TEST_F(UnionNodeTest, OrderDependenciesUnionPositionsInvalidInput) {
+  const auto od_a_to_b = OrderDependency{{_a}, {_b}};
+  const auto order_constraint = TableOrderConstraint{{ColumnID{0}}, {ColumnID{1}}};
+  _mock_node1->set_order_constraints({order_constraint});
+  EXPECT_EQ(_mock_node1->order_dependencies().size(), 1);
+
+  // Fail if inputs have different output expressions.
+  _union_node->set_right_input(_mock_node2);
+  EXPECT_THROW(_union_node->order_dependencies(), std::logic_error);
+
+  // Fail if inputs have same output expressions, but different ODs.
+  // clang-format off
+  const auto join_node_1 =
+    JoinNode::make(JoinMode::Cross,
+      _mock_node1,
+      _mock_node2);
+  const auto join_node_2 =
+    JoinNode::make(JoinMode::Inner, equals_(_a, _u),
+      _mock_node1,
+      _mock_node2);
+  // clang-format on
+
+  _union_node->set_left_input(join_node_1);
+  _union_node->set_right_input(join_node_2);
+  EXPECT_THROW(_union_node->order_dependencies(), std::logic_error);
```

Comment:
I wonder how these would be used in the optimizer. Especially since a simple check for order dependencies can throw (I would expect it returns nullopt or something).

But maybe I just don't have a good idea how to use order dependencies.

Reply:
https://github.com/hyrise/hyrise/blob/01629332003bae75fff620bfece98bdb1f962a85/src/test/lib/logical_query_plan/union_node_test.cpp#L281

Just like when we try to union tables with different columns, this currently means that we deliberately broke the query plan. Throwing should be fine.